src/kademlia/dht_storage.cpp

```diff
@@ -473,13 +477,17 @@ namespace
 
 			if (0 == m_settings.item_lifetime) return;
 
+			static auto const stime = system_clock::now();
+			static auto const ctime = aux::time_now();
+
+			auto const now = stime + (aux::time_now() - ctime);
 			time_duration lifetime = seconds(m_settings.item_lifetime);
 			// item lifetime must >= 120 minutes.
 			if (lifetime < minutes(120)) lifetime = minutes(120);
 
 			for (auto i = m_immutable_table.begin(); i != m_immutable_table.end();)
 			{
-				if (i->second.last_seen + lifetime > now)
+				if (system_clock::from_time_t(i->second.last_seen) + lifetime > now)
```

Comment:
i.e. ``if (i->second.last_seen > system_clock::now())``. you could avoid the ``to_time_t`` conversion by keeping the ``last_seen`` member a ``time_point``, and convert to and from ``time_t`` when saving and restoring.